### DISCUSSION TITLE
create Button in addColumns()

### DISCUSSION THREAD

**Discussion by yusfiadil** [Power-Components/livewire-powergrid#395] [Q&A]
how make custom button in function addColumns()  ?

![image](https://user-images.githubusercontent.com/26054562/163661858-56ad42c2-a5d8-49d6-8149-24924c032c01.png)

**Comment by yusfiadil** [comment]
And i want using function  openModal() in custom button, how ?

**Reply by luanfreitasdev** [reply]
Show what error you are getting

The commented part should work.
The Button Class will not work inside addColumns.

**Reply by BenOussama180** [reply]
sadly the commented part does not work, only the anchor "a" tag  works and gets rendered as html & i really need to renderr a button in the first button to add to cart

**Reply by luanfreitasdev** [reply]
Have you tried with Blade::render() ?

**Reply by BenOussama180** [reply]
is that a facade provided by Laravel ? never knew if that existed !! i will give a try rright now and give a feedback in the next few hourrs

**Reply by BenOussama180** [reply]
Update : 
The blade::render() does work and this is the correct way of doing it regarding @yusfiadil issue 
![Issue](https://user-images.githubusercontent.com/81111919/167860059-b4188a09-9ceb-4c52-b76d-37fdab29e98d.png)

And after some debugging it seems that  this is the correct way to pass parameters  while emiiting an event : 
![Param](https://user-images.githubusercontent.com/81111919/167863232-3775feac-3068-4745-b89d-f691c32e8d73.png)
_Hope this can help someone :)_
**However i do have a question**: when i was using buttons/rules i was switching the add to cart button to a remove from cart when the product was added to cart , any idea on how i can acheive the same logic with addColumn ? @luanfreitasdev

**Reply by pirex360** [reply]
But can you trigger the modal using Blade::render ?

**Reply by BenOussama180** [reply]
yes you can, you can do anything u want with livewire or alpine inside blade render

**Reply by pirex360** [reply]
My goal has to have something like the demo :

```
 ->openModal('delete-dish', [
                    'dishId'                  => 'id',
                    'confirmationTitle'       => 'Delete dish',
                    'confirmationDescription' => 'Are you sure you want to delete this dish?',
                ]),
```
But we cannot use Button:: on the add Column, the Blade::render is ugly but work very well....

**Reply by pirex360** [reply]
> yes you can, you can do anything u want with livewire or alpine inside blade render

How can I invoke modal like in Demo Grid, on the blade render ?

**Reply by BenOussama180** [reply]
yeap same! i wanted to put an add to cart button at the beginning of the table so its the first thing the user will see, it is ugly but i dont think there is another way, maybe in v3 they will something

**Reply by BenOussama180** [reply]
> 

i am using wire elements modal and calling it with emiting an event on click of button as shown in the picture

**Reply by pirex360** [reply]
But you changed your emit from openModal to addToCart ....  is addToCart a modal ?

**Reply by BenOussama180** [reply]
the first one is calling the wire elements modal, the second one is just passing parameters to a livewire function that does some logic

**Reply by BenOussama180** [reply]
first pic is the one that opens a modal using alpine , if you're not using alpine js , just use wire:click=$emit instead of @click=Livewire.emit

**Reply by luanfreitasdev** [reply]
Blade render works this way that I use the most:

```php
return Blade::render('<x-button cartId=\'' . $model->id . '\'/>');
```

**Reply by pirex360** [reply]
Yes, I made it work...but the modal doesn't support bootstrap....  only tailwind ! 
So I created a simple livewire with sweetalert, displaying a confirmation... that's better than a plain javascript confirm()....

On projects with tailwind, I see it will work fine, like the demo example....

**Reply by BenOussama180** [reply]
@pirex360 you can easiiy use bootstrap with wire elements modal by publishing the view and changings the classes, for a big long term project i guess its worth the time

**Reply by pirex360** [reply]
Yes, I already tried it, there is a good starting point: https://github.com/wire-elements/modal/discussions/212#discussioncomment-2737109
But with my current theme, it was a bit of CSS conflict e preferred the other approach... all I wanted is a confirm dialog modal, sweet-alert already on library used in the application, so... it was easy to implement. 

But, acording to the author, they will release a new version with bootstrap 5 support.

**Comment by BenOussama180** [comment]
Did you manage to come up with something ? 
i want to do the same ( add button of add to cart in the first column )
